taxonomy/tests/test_api.py

```diff
@@ -73,252 +60,42 @@ def setUp(self):
-        self.assertRaises(Http404, lambda: delete_vocabulary(9))
-        self.assertEquals(1, Vocabulary.objects.count())
-        self.assertRaises(ProtectedError, lambda: delete_vocabulary(
-            self.vocabulary.id))
-        delete_term(self.term.id)
-        delete_vocabulary(self.vocabulary.id)
-        self.assertEquals(0, Vocabulary.objects.count())
-
-    def test_get_learning_resources_for_term(self):
-        """
-        Test get_learning_resources_for_term
-        """
-        self.assertEquals([],
-                          list(get_learning_resources_for_term(self.term.id)))
-
-        self.learning_resource.terms.add(self.term)
-        self.assertEquals([self.learning_resource],
-                          list(get_learning_resources_for_term(self.term.id)))
-
-        self.assertRaises(Http404,
-                          lambda: get_learning_resources_for_term(9))
-
     def test_get_term(self):
         """
         Test get_term
         """
-        invalid_id = -1
-        self.assertRaises(Http404, lambda: get_term(invalid_id))
-        self.assertEquals(self.term, get_term(self.term.id))
-
-    def test_get_terms_for_learning_resource(self):
-        """
-        Test get_terms_for_learning_resource
-        """
-        self.assertEquals([],
-                          list(get_terms_for_learning_resource(
-                              self.learning_resource.id)))
-
-        self.term.learning_resources.add(self.learning_resource)
-        self.assertEquals([self.term],
-                          list(get_terms_for_learning_resource(
-                              self.learning_resource.id)))
-        self.assertRaises(Http404,
-                          lambda: get_terms_for_learning_resource(9))
-
-    def test_get_types_for_vocabulary(self):
-        """
-        Test get_types_for_vocabulary
-        """
-        self.vocabulary.learning_resource_types.add(
-            self.learning_resource_type1)
-        self.assertEquals([self.learning_resource_type1.name],
-                          list(get_types_for_vocabulary(self.vocabulary.id)))
-        self.assertRaises(Http404,
-                          lambda: get_types_for_vocabulary(3))
-
-    def test_get_vocabularies_for_type(self):
-        """
-        Test get_vocabularies_for_type
-        """
-        self.vocabulary.learning_resource_types.add(
-            self.learning_resource_type1)
-        self.assertEquals([self.vocabulary],
-                          list(get_vocabularies_for_type(
-                              self.learning_resource_type1.name)))
-        self.assertRaises(Http404,
-                          lambda: get_vocabularies_for_type("xyz"))
+        actual_term = get_term(
+            self.repo.slug,
+            self.user.id,
+            self.vocabulary.slug,
+            self.term.slug
+        )
+        self.assertEquals(self.term, actual_term)
+        with self.assertRaises(NotFound):
```

Comment:
nit for future reference:

When reading this list of tests, it would be nice to use the names of the kwargs to save the reader from constantly looking up to figure out which positional argument is missing. Don't bother to update this, though.